src/Sulu/Bundle/MediaBundle/Media/Storage/S3Storage.php

```diff
@@ -44,14 +49,16 @@ public function __construct(FilesystemInterface $filesystem, int $segments)
 
         $this->endpoint = (string) $this->adapter->getClient()->getEndpoint();
         $this->bucketName = $this->adapter->getBucket();
+
+        $this->url = null !== $url ? $url : $this->endpoint . '/' . $this->bucketName;
```

Comment:
For better readability I would put the after part in ()

```suggestion
        $this->url = null !== $url ? $url : ($this->endpoint . '/' . $this->bucketName);
```